### PR TITLE
[ci-secret-bootstrap] Remove bitwarden client and options

### PR DESCRIPTION
**Pull request by droslean**
These changes:
- Remove Bitwarden initialization via flags/ cleanup tests.
- Remove completely the bitwarden client, cleanup/refactor tests, and implement a dry run vault client that writes to a file to satisfy ci-secret-generator dryrun mode.

Follow-ups:
- Rename all Bitwarden references and variables. 

[DPTP-2170](https://issues.redhat.com/browse/DPTP-2170)

Signed-off-by: Nikolaos Moraitis <nmoraiti@redhat.com>

### CODE REVIEW

File: cmd/ci-secret-bootstrap/main_test.go [hunk 1]
```diff
@@ -1470,10 +1240,8 @@ func TestConstructSecrets(t *testing.T) {
 			},
 		},
 		{
-			name: "Secret fails when base64 decoding is requsted on invalid data",
-			bwItems: []bitwarden.Item{
-				{Name: "item", Fields: []bitwarden.Field{{Name: "key", Value: "value"}}},
-			},
+			name:  "Secret fails when base64 decoding is requsted on invalid data",
+			items: []testVaultItem{{name: "item", fields: vaultFieldsFromMap(map[string]string{"key": "value"})}},
```
Comment (alvaroaleman) [inline cmd/ci-secret-bootstrap/main_test.go:1244]: In the future, please do not change formatting like this (You did this in almost every single test case and there are a lot). Your reviewer has to review all these formatting changes for no good reason.

File: pkg/secrets/flags.go [hunk 2]
```diff
@@ -37,35 +29,19 @@ func (o *CLIOptions) Bind(fs *flag.FlagSet, getenv func(string) string, censor *
...
-func (o *CLIOptions) Validate() []error {
-	var errs []error
-	var credentialsProviderConfigured []string
-	if (o.BwUser == "") != (o.BwPasswordPath == "") {
-		errs = append(errs, fmt.Errorf("--bw-user and --bw-password-path must be specified together"))
-	} else if o.BwUser != "" {
-		credentialsProviderConfigured = append(credentialsProviderConfigured, "bitwarden")
-	}
+func (o *CLIOptions) Validate() error {
 	if o.VaultAddr != "" || o.VaultToken != "" || o.VaultTokenFile != "" || o.VaultPrefix != "" {
```
Comment (alvaroaleman) [inline pkg/secrets/flags.go:33]: We can remove this condition, vault must be configured now and the inner `if` already checks the completeness of that

File: cmd/ci-secret-bootstrap/main_test.go (hunk 1 above)
Comment (droslean) [reply cmd/ci-secret-bootstrap/main_test.go:1244]: well this is a completely different field value, but I must admin that I've changed the formating :D

File: pkg/secrets/flags.go (hunk 2 above)
Comment (droslean) [reply pkg/secrets/flags.go]: addressed.

File: cmd/ci-secret-bootstrap/main_test.go [hunk 3]
```diff
@@ -280,39 +244,37 @@ var (
 		BearerToken: "token2",
 	}
 
+	defaultConfigFrom = map[string]secretbootstrap.BitWardenContext{
```
Comment (alvaroaleman) [inline cmd/ci-secret-bootstrap/main_test.go:247]: Can you revert moving this please? Makes it harder to review and makes the tests share state

File: cmd/ci-secret-bootstrap/main_test.go [hunk 4]
```diff
@@ -1017,108 +946,60 @@ func TestValidateCompletedOptions(t *testing.T) {
 	}
 }
 
+type testVaultItem struct {
+	name         string
+	fields       []testVaultField
+	revisionTime *time.Time
+}
+
+type testVaultField struct {
```
Comment (alvaroaleman) [inline cmd/ci-secret-bootstrap/main_test.go:955]: Why are we modeling a `map[string]string]` as a `[]struct{name, value string}`?

File: cmd/ci-secret-bootstrap/main_test.go [hunk 5]
```diff
@@ -1017,108 +946,60 @@ func TestValidateCompletedOptions(t *testing.T) {
 	}
 }
 
+type testVaultItem struct {
```
Comment (alvaroaleman) [inline cmd/ci-secret-bootstrap/main_test.go:949]: Why are we creating new types instead of using pkg/vaultclient.KVData?

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**